torch/_refs/linalg/__init__.py

```diff
@@ -151,6 +151,23 @@ def vector_norm(
         reduce_sum = partial(torch.sum, dim=dim, keepdim=keepdim)
 
         is_ord_even = ord % 2 == 0 if isinstance(ord, IntLike) else ord % 2.0 == 0.0
+        if (dim is None and guard_size_oblivious(x.numel() == 1)) or (
+            dim is not None
+            and (x.ndim > 0 and all(guard_size_oblivious(x.shape[d] == 1) for d in dim))
+        ):
+            x = torch.abs(x)
+            if x.ndim > 64:
```

Comment:
NIT: We should apply this check before we take the absolute value. No sense in doing the work if we are failing anyway.